test/cri-containerd/removepodsandbox_test.go

```diff
@@ -0,0 +1,171 @@
+// +build functional
+
+package cri_containerd
+
+import (
+	"context"
+	"testing"
+
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+)
+
+type TestConfig struct {
+	name             string
+	containerName    string
+	requiredFeatures []string
+	runtimeHandler   string
+	sandboxImage     string
+	containerImage   string
+	cmd              []string
+}
+
+func getRunPodSandboxRequest(t *testing.T, runtimeHandler string) runtime.RunPodSandboxRequest {
+	return runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name(),
+				Uid:       "0",
+				Namespace: testNamespace,
+			},
+		},
+		RuntimeHandler: runtimeHandler,
+	}
+}
+
+// Utility function to run tests for a sandbox pod without any containers
+func runPodSandboxTestWithoutPodStop(t *testing.T, request *runtime.RunPodSandboxRequest) {
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	podId := runPodSandbox(t, client, ctx, request)
+	removePodSandbox(t, client, ctx, podId)
+
+	r, _ := client.PodSandboxStatus(ctx, &runtime.PodSandboxStatusRequest{
+		PodSandboxId: podId,
+	})
+
+	if r != nil {
+		t.Fatalf("Pod should've been removed %s", r)
+	}
+}
+
+// Utility function to run tests for a sandbox pod with and a single container inside the pod
+func runContainerInSandboxTest(t *testing.T, tc *TestConfig) {
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	annotations := map[string]string{
+		"io.microsoft.virtualmachine.computetopology.memory.allowovercommit": "true",
+	}
+
+	request := getRunPodSandboxRequest(t, tc.runtimeHandler)
+
+	podId := runPodSandbox(t, client, ctx, &request)
+	containerId := createContainerInSandbox(t, client, ctx, podId, tc.containerName, tc.containerImage, tc.cmd, annotations, []*runtime.Mount{}, request.Config)
+
+	startContainer(t, client, ctx, containerId)
+
+	removePodSandbox(t, client, ctx, podId)
+
+	containerStatus, _ := client.ContainerStatus(ctx, &runtime.ContainerStatusRequest{
+		ContainerId: containerId,
+	})
+
+	if containerStatus != nil {
+		t.Fatalf("Container in pod %s should've been stopped and removed, but container status is %s", podId, containerStatus)
+	}
```

Comment:
Same as the other comment about checking for error. We could also check that there is no containerStatus though.